src/Areas/Marketplace/Commands/Product/ProductGetCommand.cs

```diff
@@ -0,0 +1,187 @@
+        options.PricingAudience = parseResult.GetValueForOption(_pricingAudienceOption);
+        options.ObjectId = parseResult.GetValueForOption(_objectIdOption);
+        options.AltSecId = parseResult.GetValueForOption(_altSecIdOption);
+        return options;
+    }
+
+    [McpServerTool(
+        Destructive = false,
+        ReadOnly = true,
+        Title = CommandTitle)]
+    public override async Task<CommandResponse> ExecuteAsync(CommandContext context, ParseResult parseResult)
+    {
+        var options = BindOptions(parseResult);
+
+        try
+        {
+            // Required validation step
+            if (!Validate(parseResult.CommandResult, context.Response).IsValid)
+            {
+                return context.Response;
+            }
+
+            // Get the marketplace service from DI
+            var marketplaceService = context.GetService<IMarketplaceService>();
+
+            // Call service operation with required parameters
+            var result = await marketplaceService.GetProduct(
+                options.ProductId!,
+                options.Subscription!,
+                options.IncludeStopSoldPlans,
+                options.Language,
+                options.Market,
+                options.LookupOfferInTenantLevel,
+                options.IncludeHiddenPlans,
+                options.PlanId,
+                options.SkuId,
+                options.IncludeServiceInstructionTemplates,
+                options.PartnerTenantId,
+                options.PricingAudience,
+                options.ObjectId,
+                options.AltSecId,
+                options.Tenant,
+                options.RetryPolicy);
+
+            // Set results
+            context.Response.Results = result != null ?
+                ResponseResult.Create(
+                    new ProductGetCommandResult(result),
+                    MarketplaceJsonContext.Default.ProductGetCommandResult) :
+                null;
+        }
+        catch (Exception ex)
+        {
+            // Log error with all relevant context
+            _logger.LogError(ex,
+                "Error getting marketplace product. ProductId: {ProductId}, Subscription: {Subscription}, Options: {@Options}",
+                options.ProductId, options.Subscription, options);
+            HandleException(context.Response, ex);
+        }
+
+        return context.Response;
+    }
+
+    // Implementation-specific error handling
+    protected override string GetErrorMessage(Exception ex) => ex switch
+    {
+        HttpRequestException httpEx when httpEx.Message.Contains("404") =>
+            "Marketplace product not found. Verify the product ID exists and you have access to it.",
+        HttpRequestException httpEx when httpEx.Message.Contains("403") =>
+            "Authorization failed accessing the marketplace product. Verify your permissions and subscription access.",
+        HttpRequestException httpEx when httpEx.Message.Contains("401") =>
```

Comment:
We shouldn't have auth messages like az login, because we use DAC any of those auth options would work.  I believe we handle auth messages at a global level.